### PR TITLE
Tests/881, Add tests for secretary service

### PR DESCRIPTION
**Pull request by BohdanZaliubovskyi**
Related issue: #881 
Added test coverage for class CharlieBackend.Business\Services\SecretaryService
Added ICloneable interface for classes: Account, Attachement, BaseEntity, Secretary, SecretaryDto, UpdateSecretaryDto

**Test coverage results:**
**Before:**
image
image

**After:**
image
image

### CODE REVIEW

File: CharlieBackend.Core/Entities/Account.cs [hunk 1]
```diff
@@ -39,5 +43,26 @@ public partial class Account : BaseEntity
...
+
+                try
+                {
+                    binFormatter.Serialize(tempStream, this);
+                    tempStream.Seek(0, SeekOrigin.Begin);
+
+                    account = binFormatter.Deserialize(tempStream) as Account;
+                }
+                catch(Exception ex)
+                { }
```
Comment (PavelTsyposhchuk) [inline CharlieBackend.Core/Entities/Account.cs]: not used ex

**Review by PavelTsyposhchuk — CHANGES_REQUESTED** [review]

File: CharlieBackend.Core/Entities/Account.cs (hunk 1 above)
Comment (BohdanZaliubovskyi) [reply CharlieBackend.Core/Entities/Account.cs]: Ty, it's just testing code.

**Review by PavelTsyposhchuk — APPROVED** [review]

**Review by BobTheSuperman — APPROVED** [review]

File: CharlieBackend.Api.UnitTes/SecretaryServiceTests.cs [hunk 2]
```diff
@@ -16,62 +20,472 @@ public class SecretaryServiceTests : TestBase
         private readonly Mock<IAccountService> _accountServiceMock;
         private readonly IMapper _mapper;
         private readonly Mock<INotificationService> _notificationServiceMock;
+        private readonly SecretaryService _secretaryService;
+        Mock<ISecretaryRepository> _secretaryRepositoryMock;
```
Comment (apyura) [inline CharlieBackend.Api.UnitTes/SecretaryServiceTests.cs]: please, add access modifier

**Review by apyura — APPROVED** [review]

File: CharlieBackend.Api.UnitTes/SecretaryServiceTests.cs (hunk 2 above)
Comment (BohdanZaliubovskyi) [reply CharlieBackend.Api.UnitTes/SecretaryServiceTests.cs]: Done

File: CharlieBackend.Api.UnitTes/SecretaryServiceTests.cs [hunk 3]
```diff
@@ -16,62 +20,472 @@ public class SecretaryServiceTests : TestBase
...
+        }
+
+        private Secretary GetSecretaryWithIdAndAccountId()
+        {
+            return new Secretary()
+            {
+                Id = _secretaryExpectedId,
+                AccountId = _accountExpectedId
+            };
+        }
```
Comment (apyura) [inline CharlieBackend.Api.UnitTes/SecretaryServiceTests.cs:49]: please, add newline between methods

File: CharlieBackend.Api.UnitTes/SecretaryServiceTests.cs [hunk 4]
```diff
@@ -16,62 +20,472 @@ public class SecretaryServiceTests : TestBase
...
+                .Should()
+                .BeEquivalentTo(ErrorCode.Conflict);
+        }
+
+        [Fact]
+        public async Task UpdateSecretaryAsync_ExceptionDuringUpdatingSecretary_ShouldReturnErrorCodeInternalServerError()
+        {
+            //Arrange
+            _unitOfWorkMock.Setup(x => x.SecretaryRepository)
+                .Throws(new Exception());
```
Comment (apyura) [inline CharlieBackend.Api.UnitTes/SecretaryServiceTests.cs]: Bad practice: exceptions throwing of base type

**Review by apyura — CHANGES_REQUESTED** [review]

File: CharlieBackend.Api.UnitTes/SecretaryServiceTests.cs (hunk 4 above)
Comment (BohdanZaliubovskyi) [reply CharlieBackend.Api.UnitTes/SecretaryServiceTests.cs]: Ok.
In code we have next situation:
```C#
catch
{
    _unitOfWork.Rollback();

    return Result<SecretaryDto>.GetError(ErrorCode.InternalServerError,
          "Cannot update secretary.");
}
```
We don't have any exception object and any type of exception, so what type of exception should i throw ?

File: CharlieBackend.Api.UnitTes/SecretaryServiceTests.cs (hunk 3 above)
Comment (BohdanZaliubovskyi) [reply CharlieBackend.Api.UnitTes/SecretaryServiceTests.cs:49]: Done.

**Review by Ssashaa — APPROVED** [review]

**Review by apyura — APPROVED** [review]

**Review by andy-dendy-repo — APPROVED** [review]

**Review by DevPew01 — APPROVED** [review]

**Pull request merged**